cmd/operator-sdk/generate/packagemanifests/cmd.go

```diff
@@ -0,0 +1,138 @@
+
+	kbutil "github.com/operator-framework/operator-sdk/internal/util/kubebuilder"
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+)
+
+//nolint:maligned
+type packagemanifestsCmd struct {
+	// Options to turn on different parts of packaging.
+	kustomize bool
+	manifests bool
+
+	// Common options.
+	operatorName string
+	version      string
+	fromVersion  string
+	inputDir     string
+	outputDir    string
+	deployDir    string
+	apisDir      string
+	crdsDir      string
+	updateCRDs   bool
+	stdout       bool
+	quiet        bool
+
+	// Interactive options.
+	interactiveLevel projutil.InteractiveLevel
+	interactive      bool
+
+	// Package manifest options.
+	channelName      string
+	isDefaultChannel bool
+}
+
+// NewCmd returns the 'packagemanifests' command configured for the new project layout.
+func NewCmd() *cobra.Command {
+	c := &packagemanifestsCmd{}
+
+	cmd := &cobra.Command{
+		Use:   "packagemanifests",
+		Short: "Generates a package manifests format",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 0 {
+				return fmt.Errorf("command %s doesn't accept any arguments", cmd.CommandPath())
+			}
+
+			// Check if the user has any specific preference to enable/disable
+			// interactive prompts. Default behaviour is to disable the prompt
+			// unless a base package does not exist.
+			if cmd.Flags().Changed("interactive") {
+				if c.interactive {
+					c.interactiveLevel = projutil.InteractiveOnAll
+				} else {
+					c.interactiveLevel = projutil.InteractiveHardOff
+				}
+			}
+
+			// Generate kustomize bases and manifests by default if no flags are set
+			// so the default behavior is "do everything".
+			fs := cmd.Flags()
+			if !fs.Changed("kustomize") && !fs.Changed("manifests") {
+				c.kustomize = true
+				c.manifests = true
+			}
+
+			cfg, err := kbutil.ReadConfig()
+			if err != nil {
+				log.Fatal(fmt.Errorf("error reading configuration: %v", err))
+			}
+			c.setCommonDefaults(cfg)
+
+			if c.kustomize {
+				if err = c.runKustomize(cfg); err != nil {
+					log.Fatalf("Error generating package bases: %v", err)
+				}
+			}
+			if c.manifests {
+				if err = c.validateManifests(); err != nil {
+					return fmt.Errorf("invalid command options: %v", err)
+				}
+				if err = c.runManifests(cfg); err != nil {
+					log.Fatalf("Error generating package manifests: %v", err)
+				}
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().BoolVar(&c.kustomize, "kustomize", false, "Generate kustomize bases")
+	cmd.Flags().BoolVar(&c.manifests, "manifests", false, "Generate package manifests")
+	cmd.Flags().BoolVar(&c.stdout, "stdout", false, "Write package to stdout")
+
+	c.addCommonFlagsTo(cmd.Flags())
+
+	return cmd
+}
+
+func (c *packagemanifestsCmd) addCommonFlagsTo(fs *pflag.FlagSet) {
+	fs.StringVar(&c.operatorName, "operator-name", "", "Name of the packaged operator")
+	fs.StringVarP(&c.version, "version", "v", "", "Semantic version of the packaged operator")
```

Comment:
Just to test the unhappy path of `operator-sdk generate version` I'm not sure where we're checking the version flag but we could make the error a bit clearer:
```console
$ operator-sdk generate packagemanifests
Generating package manifests kustomize bases

Display name for the operator (required):
> FooBar
...

Bases generated successfully in config/packages
Error: invalid command options:  is not a valid semantic version: Version string empty

Usage:
  operator-sdk generate packagemanifests [flags]

Flags:
      --apis-dir string        Root directory for API type defintions
      --channel string         Channel name for the generated package
...
```

Granted it's obvious to folks familiar with `generate csv` but just looking at this error I still don't know the exact flag I need to specify.
Maybe we can verify the `--version` flag a bit earlier to error out and say it's required.